tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,272 @@
+from ims.common.log import trace
+from pytest import set_trace as bp
+import logging
+
+_cfg = config.get()
+
+PROJECT = _cfg.tests.project
+NETWORK = _cfg.tests.network
+
+EXIST_IMG_NAME = _cfg.tests.exist_img_name
+NEW_SNAP_NAME = _cfg.tests.new_snap_name
+NOT_EXIST_IMG_NAME = _cfg.tests.not_exist_img_name
+NOT_EXIST_SNAP_NAME = _cfg.tests.not_exist_snap_name
+
+HIL_CALL_TIMEOUT = 5  # constants.HIL_CALL_TIMEOUT
+
+POOL = _cfg.fs.pool
+# test image
+CEPH_IMG = "BMI_TEST_CEPH"
+# test image size
+CEPH_IMG_SIZE = 1024
+# test snap_image
+CEPH_EXISTING_IMG = "bmi_ci.img"
+CEPH_SNAP_IMG = "BMI_TEST_CEPH_SNAP"
+# test write
+TEST_DATA = 'bmi' * 10
+OFFSET = 0
+# test clone
+CEPH_CHILD_IMG = "BMI_TEST_CEPH_CHILD_IMAGE"
+
+
+class TestListImages(unittest.TestCase):
+    """
+    Test list images
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        imgs = self.fs.list_images()
+        command = "echo {0} | sudo -S rbd --keyring {1} --id {2} ls " \
+                  "{3}".format(self.fs.password, self.fs.keyring,
+                               self.fs.rid, self.fs.pool)
+        p = subprocess.Popen(command, shell=True, stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+        output, err = p.communicate()
+        rbd_imgs = output.split()
+        self.assertListEqual(imgs, rbd_imgs)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def tearDown(self):
+        print('In tearDown()')
+        time.sleep(HIL_CALL_TIMEOUT)
+
+
+class TestCreateImage(unittest.TestCase):
+    """
+    Test create image
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        imgs = self.fs.list_images()
+        self.assertIn(CEPH_IMG, imgs)
+
+    def tearDown(self):
+        self.fs.remove(CEPH_IMG)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+
+class TestRemoveImage(unittest.TestCase):
+    """
+    Test remove image and remove snapshot
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertRaises(file_system_exceptions.ImageHasSnapshotException,
+                          self.fs.remove, CEPH_IMG)
+        removess = self.fs.remove_snapshot(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(removess)
+        remove = self.fs.remove(CEPH_IMG)
+        self.assertTrue(remove)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_IMG, imgs)
+
+    def tearDown(self):
+        pass
+
+
+class TestClone(unittest.TestCase):
+    """
+    Clone image
+    """
```

Comment:
about docstrings:

* if it's small, just have a 1 line docstring `"""Clone image"""`
* but in this case I think you should write what exactly you are testing here. `test to see if snapshots are protected in ceph"""` or something like that.